crates/tabby/tests/goldentests.rs

```diff
@@ -16,9 +15,12 @@ lazy_static! {
             .arg("TabbyML/StarCoder-1B")
             .arg("--port")
             .arg("9090")
-            .arg("--device")
-            .arg("metal")
             .kill_on_drop(true);
```

Comment:
refactor above block into a function, instead of a static variable.

Thus we will have freedom to pass in different device enum in different test.